### PR TITLE
deps: make rate errors non-redactable

### PR DESCRIPTION
**Pull request by knz**
(NB: the solution taken here is a bit too manual for my taste. OK to do this once, but we need a more general mechanism. Tracking that problem in #104014.)

Fixes CRDB-27646.
Epic: CRDB-27642.

From: https://github.com/cockroachlabs/support/issues/2284
Depends on https://github.com/cockroachlabs/crl-infrastructure/pull/860 and https://github.com/cockroachdb/x-time/compare/master...20230525-crdb-errors.

Before this patch:
```
I230430 04:06:26.955997 7563189786 kv/kvserver/replicate_queue.go:312⋮ [T1,n3,replicate,s3,r156/3:‹×›] 1757110 ‹×›
```

After:
```
I230430 04:06:26.955997 7563189786 kv/kvserver/replicate_queue.go:312⋮ [T1,n3,replicate,s3,r156/3:‹×›] 1757110 rate: Wait(n=1) would exceed context deadline
```

Release note: None

### CODE REVIEW

**Review by arulajmani — APPROVED** [review]
I'm stamping this based on the before and after output; I couldn't tell you how this works -- the more general issue you linked there provides a hint:

```
forking the upstream dependencies and manually rewrite calls to errors.New and fmt.Errorf to reuse the crdb errors' library errors.New and errors.Newf 
```

I can't tell where/how this is happening. Could you point me to it, or ELI5 what this PR is doing, for my edification? 

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/103886#-:-NWrCqAc94_hhSbQzUnS:b-jlkqwi)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz)

___
*[`-- commits` line 6 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/103886#-NWrB5sx27l1QHzbVE9M):*
nit: should we update the commit message to match the PR description before merging?

**Review by abarganier — APPROVED** [review]
+1 to what @arulajmani said - LGTM based on the PR description, but this is also the first time I'm seeing this type of control over redaction. I'd also appreciate seeing some more info about how this works, mostly for my own curiosity & to better my knowledge of redaction!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/103886#-:-NWrFtvH3nZWZOr-0xis:b8cnidg)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz)

**Comment by knz** [comment]
>  I can't tell where/how this is happening

The change to go.mod should have put you on the right path.
The diff is here: 
https://github.com/cockroachdb/x-time/compare/master...20230525-crdb-errors

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/103886)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani)

___
*[`-- commits` line 6 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/103886#-NWrB5sx27l1QHzbVE9M):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

nit: should we update the commit message to match the PR description before merging? 
</blockquote></details>

Done.

**Comment by knz** [comment]
bors r=abargainier,arulajmani
